upm: Fix for building python bindings.

pylbert opened the pull request:
Updated to the latest tagged version of UPM.

```
* Fix for building python2 vs python3 bindings
* Added rdepends on mraa-java to upm recipe
```

Signed-off-by: Noel Eck noel.eck@intel.com

noel-eck committed:
upm: Fix for building python bindings.

Updated to the latest tagged version of UPM.

    * Fix for building python2 vs python3 bindings
    * Added rdepends on mraa-java to upm recipe

Signed-off-by: Noel Eck <noel.eck@intel.com>

rojkov commented:
test this please

rojkov commented:
ok to test

kad commented on recipes-devtools/upm/upm_0.7.1.bb:
```diff
@@ -16,16 +16,14 @@ inherit distutils-base pkgconfig python-dir cmake
 
 FILES_${PN}-doc += " ${datadir}/upm/examples/"
-RDEPENDS_${PN} += " mraa"
+RDEPENDS_${PN} += " mraa mraa-java"
```
probably would be good to be masked by java distro feature ?
@pohly, what do you think ?

pohly replied:
There is no standard "java" distro feature, and thus meta-intel-iot-middleware cannot depend on it.

We recently introduced one in Ostro (https://github.com/ostroproject/meta-ostro/pull/216) and would have to adapt that to this change.

But I have a different comment: wouldn't it be better to split out the different language bindings into separate packages (like upm-java, upm-python, etc.)? Then even for the case where the distro has Java support enabled (like regular Ostro OS), images can still be created where "normal" upm gets installed without pulling in Java.

tripzero replied:
@pohly We are splitting them out into different packages.  

@pylbert The above line is wrong because mraa-java is an optional runtime-dependency.  We should either make it conditional based on whether or not the upm-java bindings are enabled, or just remove it all together.

pylbert replied:
@tripzero it's not an optional runtime dependency to the upm package as the recipe is today.  If the mraa-java package is NOT built and the upm package is configured to build java-upm, the upm build will fail.

tripzero replied:
@pylbert We need to check if BINDINGS_${PN}-mraa contains "java" before making it a runtime dependency.  If java-mraa is disabled by user, upm will fail also.  So it needs to be something like this:

RDEPENDS_${PN} += "${@bb.utils.contains('BINDINGS_pn-mraa', 'java', 'java-mraa', '', d)}"

@bavery22 is this correct ?

rojkov commented:
test this please (second attempt)

kad commented:
ok to test

tripzero commented on recipes-devtools/upm/upm_0.7.1.bb:
```diff
@@ -16,16 +16,14 @@ inherit distutils-base pkgconfig python-dir cmake
-#BINDINGS ?= "python nodejs java"
-BINDINGS ?= "nodejs java"
+# Build all language bindings by default
+BINDINGS ?= "python nodejs java"
```
This re-enables python, but how does it actually fix the build issue?

pylbert replied:
@tripzero the rename from 0.7.0 -> 0.7.1 fixes the python binding build issue.

tripzero replied:
The python issue is random, so I'm not surprised it would build now with this tiny change.  I suspect the build issue will return sometime in the future though.

tripzero replied:
Also, there's a note on line 26 that should not be removed.

pylbert replied:
@tripzero the recipe rename pulls the 0.7.1 tagged version. The new UPM version (0.7.1) defines which python version (2 vs 3) to use as a hint to find_package PythonLibs  

I will rebase my PR to NOT add the mraa-java rdepends since this seems to be blocking the python binding fix from getting merged.

I'm not sure about the note on line 26.

tripzero replied:
ah, okay.  That makes sense.  I'll look for the new PR and merge ASAP

noel-eck committed:
upm: Removed rdepends for mraa-java package

Backing out the mraa-java change so the switch from 0.7.0 -> 0.7.1
version of upm can be merged.

Signed-off-by: Noel Eck <noel.eck@intel.com>

tripzero commented:
lgtm.  We'll wait for OstroProjectCI to finish building and merge.  Thanks
